pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -710,12 +713,38 @@ func (gce *GCECloud) EnsureLoadBalancer(clusterName string, apiService *api.Serv
 		glog.Infof("EnsureLoadBalancer(%v(%v)): deleted forwarding rule", loadBalancerName, serviceName)
 	}
 	if tpExists && tpNeedsUpdate {
-		if err := gce.deleteTargetPool(loadBalancerName, gce.region); err != nil {
+		hc := []*compute.HttpHealthCheck{}
+		if path := getServiceHealthCheckPath(apiService); path != "" {
+			healthCheckNodePort := getServiceHealthCheckNodePort(apiService)
```

Comment:
maybe we can encapsulate this repeated 2 step dance (getPath/getNodePort) into a single function that returns a struct with path,health check?